packages/video_player/video_player_web/lib/video_player_web.dart

```diff
@@ -176,10 +176,38 @@ class _VideoPlayer {
         isInitialized = true;
         sendInitialized();
       }
+      if (isBuffering) {
+        isBuffering = false;
+        eventController.add(VideoEvent(eventType: VideoEventType.bufferingEnd));
+      }
```

Comment:
Looks like this part is repeated several times, can we refactor this into a function? Maybe something like:
`sendBufferingEndEventIfNeeded()`

Reply:
This looks like a good opportunity for a isBuffering setter, actually:

```dart
void setBuffering(bool buffering) {
  if (isBuffering != buffering) {
    isBuffering = buffering;
    eventController.add(VideoEvent(eventType: isBuffering ? VideoEventType.bufferingStarted : VideoEventType.bufferingEnd));
  }
}
```

(However there's some special cases like the onError that break the nice pattern :P)

Reply:
Both good ideas, I chose the `setBuffering` approach in the end that @ditman mentioned (on both Android & Web) - and error events on both these platforms will now also send the buffering end event also (only if was buffering previously) - made sense to send this as otherwise there could be a potential for an invalid buffering state when the player errors.